### PR TITLE
Cutting onions makes you cry

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
Kind of a feature, kind of a fix? Apparently we had this reagent that's supposed to make you cry and is from onions, but that reagent was only in red onions, and also it didn't work. This fixes tear juice, and adds it to onions.

Now when you cut onions it will release tear juice smoke, which will blur your vision and make you *cry. 

## Why It's Good For The Game
Having a big brown cloud EXPLODE from onions but also not do anything was kind of weird. Also I burned my eyes cutting onions the other day and now chef mains must also suffer.

## Changelog

:cl:
fix: cutting onions actually makes you cry (better put some goggles on!)
/:cl:

### CODE REVIEW

**Comment by VladinXXV** [comment]
I'd like to suggest giving chefs immunity to this, actually. Hear me out. Or don't, I can't make you. When you cut an onion, the chemical that makes your eyes sting is released when you crush the cells. This happens in large amounts due to two factors:

- A dull blade

- Cutting against the "grain"

When cutting an onion, if you cut in the controversial "longitude" (top to bottom) rather than "latitude", you will rupture fewer cells. This is what I mean when I talk about the "grain". As for a dull blade, when your knife isn't properly sharpened, you actually tend to "crush" things rather than "slice" them. The difference is subtle, but since we're talking about rupturing cell walls here, the difference matters. A sharper blade slices instead of crushes. By combining a well maintained blade with good technique, you can minimize the amount of this chemical released, greatly reducing if not entirlely eliminating the "crying" part of cutting onions. This also changes the flavor of the onion, since you lose less of that "onion chemical".

I see two ways this could be implemented. Sharpened knives (via whetstone) could reduce/eliminate the effect. Having the chef skillchip could serve as a substitute. Or both could be implemented.

I can understand if this is outside the scope of the PR, or something you aren't interested in doing in general, but it's a neat little bit of information I have for some reason and I thought it could be interesting to consider.

**Comment by dragomagol** [comment]
I definitely appreciate your passion for it! I didn't intend to add much more beyond the actual crying emote, but if you want to take it up after this then by all means.

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 1]
```diff
@@ -468,6 +468,28 @@
...
+	. = ..()
+	if(!ishuman(exposed_mob))
+		return
+
+	var/mob/living/carbon/victim = exposed_mob
+	if(methods & (INGEST | VAPOR))
+		var/tear_proof = victim.is_eyes_covered()
+
+		//check for protection
+		if (!(tear_proof))
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:488]:
```suggestion
		if (!tear_proof)
```

(or cut the variable entirely)

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 2]
```diff
@@ -468,6 +468,28 @@
...
+/datum/reagent/consumable/tearjuice/expose_mob(mob/living/exposed_mob, methods = INGEST, reac_volume)
+	. = ..()
+	if(!ishuman(exposed_mob))
+		return
+
+	var/mob/living/carbon/victim = exposed_mob
+	if(methods & (INGEST | VAPOR))
+		var/tear_proof = victim.is_eyes_covered()
+
+		//check for protection
```
Comment (Mothblocks) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm:487]: useless comment?

**Pull request merged**